scrapy/settings/__init__.py

```diff
@@ -88,19 +103,48 @@ def getdict(self, name, default=None):
             value = json.loads(value)
         return dict(value)
 
+    def _getcomposite(self, name):
+        # DO NOT USE THIS FUNCTION IN YOUR CUSTOM PROJECTS
+        # It's for internal use in the transition away from the _BASE settings and
+        # will be removed along with _BASE support in a future release
+        basename = name + "_BASE"
+        if basename in self.attributes:
+            warnings.warn('_BASE settings are deprecated.',
+                          category=ScrapyDeprecationWarning)
+            compsett = BaseSettings(self[name + "_BASE"], priority='default')
+            compsett.update(self.get(name))
```

Comment:
True. I forgot that `__getitem__()` doesn't throw `KeyError`s either. I like all your nitpicks ;)